mlflow/cli/__init__.py

```diff
@@ -676,28 +714,9 @@ def fetch_runs(token=None):
 
         run_ids.extend([run.info.run_id for run in fetch_runs()])
 
-    for run_id in set(run_ids):
-        run = backend_store.get_run(run_id)
-        if run.info.lifecycle_stage != LifecycleStage.DELETED:
-            raise MlflowException(
-                f"Run {run_id} is not in `deleted` lifecycle stage. Only runs in"
-                " `deleted` lifecycle stage can be deleted."
-            )
-        # raise MlflowException if run_id is newer than older_than parameter
-        if older_than and run_id not in deleted_run_ids_older_than:
-            raise MlflowException(
-                f"Run {run_id} is not older than the required age. "
-                f"Only runs older than {older_than} can be deleted.",
-                error_code=INVALID_PARAMETER_VALUE,
-            )
-        # raise MlflowException if run_id is newer than older_than parameter
-        if older_than and run_id not in deleted_run_ids_older_than:
-            raise MlflowException(
-                f"Run {run_id} is not older than the required age. "
-                f"Only runs older than {older_than} can be deleted.",
-                error_code=INVALID_PARAMETER_VALUE,
-            )
-        artifact_repo = get_artifact_repository(run.info.artifact_uri)
+    def _delete_with_artifacts(artifact_uri, delete_fn, entity_desc):
```

Comment:
Probably we don't need to combine artifact deletion, entity deletion and description display and simply define a function for artifact deletion only.